streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -650,6 +650,25 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
   final def concat[R1 <: R, E1 >: E, A1 >: A](other: => ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
     ZStream(UIO.succeed(self), UIO(other)).flatMap(ZStream.unwrap)
 
+  /**
+   * Emits elements of this stream with a fixed delay in between, regardless of how long it
+   * takes to produce a value.
+   */
+  final def delay[R1 <: R, E1 >: E, A1 >: A](duration: Duration): ZStream[R1 with Clock, E1, A1] =
```

Comment:
Nice 👍

Reply:
Could this be implemented as `stream.spaced(Schedule.fixed(duration))`?

Reply:
I had tried that exact version initially but noticed the schedule would apply to each element of the stream, in this case repeat each element infinitely. I'll try with `stream.spaced(Schedule.fixed(duration) && Schedule.once)` 🤔

Reply:
Ah right, yes, that should do the trick!

Reply:
Hmm, I don't think this is going to work. That will repeat once, so each element is repeated once, but also emit the schedule's output, so I am getting:
```scala
scala> Stream(1,2,3).spaced((ZSchedule.fixed(1.second) && Schedule.once).unit).runCollect.unsafeRun
res3: List[AnyVal] = List(1, 1, (), 2, 2, (), 3, 3, ())
```
IIUC also this will behave as follows:
```
1===1()2===2()3===3()
```
whereas what I'm looking for is
```
1===2===3
```
I don't think it is possible to express with `spaced`.

Reply:
Right, `spaced` will always add an additional element if there's a delay.

Reply:
@regiskuckaertz @iravid Do we need a new version of `spaced` that doesn't? 

This `delay` combinator seems like more common use case and it'd be nice if we could have a way to get all such delays (e.g. `fixed`), even if less common and not rising to the level of a separate specialization.

Reply:
Yes, I didn't understand `ZStream.spaced` at first because I was trying to map it to `ZSchedule.spaced` which seemed to do what I wanted.

Reply:
Can this be implemented in terms of `throttleShape`?